Hy vendors api

HristoYonkov opened the pull request:
1. Create new Vendor api functionality added.
2. Edit  Vendor api functionality added.
3. Delete Vendor api functionality added.

![image](https://github.com/user-attachments/assets/4f18d8e1-f77b-481c-898c-a048e2d2ecc5)

HristoYonkov committed:
make the field accept only numbers!

HristoYonkov committed:
Update bg.json

HristoYonkov committed:
add Vendor Mutation function for adding new vendor

HristoYonkov committed:
change vendorNumber to string

HristoYonkov committed:
update data table to accept new props

HristoYonkov committed:
integrate react querry to Vendors table (not finished)

HristoYonkov committed:
Update VendorsTable.tsx

HristoYonkov committed:
add cChipList component to the table

HristoYonkov committed:
update the warning string on delete

HristoYonkov committed:
add interface and proper translate for warning dialog

HristoYonkov committed:
add delete functionality

HristoYonkov committed:
remove the comments and fix the suspenseQuery bug

HristoYonkov committed:
Update bg.json

HristoYonkov committed:
addd new field for markers

HristoYonkov committed:
remove comment

HristoYonkov committed:
add markers to the submit hadnler

HristoYonkov committed:
add markers to the validation scheme

HristoYonkov committed:
add console to see the form for debugging

HristoYonkov committed:
add markers to the vendorFormDto

Hristo-Boyadzhiev committed:
Update Zones.tsx

Hristo-Boyadzhiev committed:
Boyadzhiev changes

HristoYonkov committed:
back the code

HristoYonkov committed:
back old code

HristoYonkov committed:
update form for edit, to accept default values

HristoYonkov committed:
add update modal, form and logic for updating record

HristoYonkov committed:
fix the bug for showing proper button name when not using stepper form

HristoYonkov committed:
Update bg.json

HristoYonkov committed:
add dynamic name for snackbar

HristoYonkov committed:
update swagger

HristoYonkov committed:
update models

HristoYonkov committed:
update the form to accept defaultValues and visualize markers

HristoYonkov committed:
use markers and add them to update the form

HristoYonkov committed:
Create useDeleteMarker.ts

HristoYonkov committed:
Create useGetMarkers.ts

HristoYonkov committed:
Create usePostMarker.ts

HristoYonkov committed:
Create useUpdateMarker.ts

HristoYonkov committed:
Create useDeleteZone.ts

HristoYonkov committed:
Create useGetZones.ts

HristoYonkov committed:
Create usePostZone.ts

HristoYonkov committed:
Create useUpdateZone.ts

HristoYonkov committed:
change name of interface and add console for debugging

HristoYonkov committed:
cast markerIds array to numeric values as BE needed

HristoYonkov committed:
remove the consoles

HristoYonkov committed:
Create DeliveriesTableActionsMenu.tsx

HristoYonkov committed:
Create VendorTableActionsMenu.tsx

HristoYonkov committed:
Create ZonesTableActionsMenu.tsx

HristoYonkov committed:
switch imports

HristoYonkov committed:
remove old files

HristoYonkov committed:
update imports

HristoYonkov committed:
update imports

HristoYonkov committed:
Merge branch 'main' into HY-Vendors_API

vexleet commented on src/components/features/actionsMenu/VendorTableActionsMenu.tsx:
```diff
@@ -3,34 +3,100 @@ import Menu from '@mui/material/Menu'
 
+  const queryClient = useQueryClient()
+
+  const mutationDelete = useMutation({
```
you can check out what @Hristo-Boyadzhiev has done with the zones in `hooks/services/zones` and you can move the query hooks and make them like what he has done to make things simpler and easier to read.

HristoYonkov replied:
It is fixed now, also i call TableActionsMenu to build dynamicly buttons for actions menu

vexleet commented on src/components/features/actionsMenu/VendorTableActionsMenu.tsx:
```diff
@@ -53,17 +119,39 @@ export default function VendorTableActionsMenu({
         ))}
       </Menu>
 
+      {selectedOption === 'Редактирай' && (
```
dont have options in Bulgarian. make them in English. Like what you have done with the "move entry" one

HristoYonkov replied:
Fixed.

vexleet commented on src/components/features/forms/NewVendorForm.tsx:
```diff
@@ -47,13 +62,30 @@ export default function NewVendorForm({
+              value={field.value || []}
+              onChange={(e) => field.onChange(e.target.value)}
+              renderValue={(selected) => (selected as string[]).join(', ')}>
+              {markers.map((marker: MarkerDto) => (
+                <MenuItem key={marker.id} value={marker.id?.toString()}>
+                  <ListItemText primary={marker.name} />
+                </MenuItem>
```
we need to show the marker name and not id when we have selected something
![image](https://github.com/user-attachments/assets/98561a1f-6e7e-407b-95ff-a82df14d44e6)

You can check what Boyadzhiev has done in Zones Form to fix that.

HristoYonkov replied:
Fixed.

vexleet committed:
fix issues after merge conflicts

HristoYonkov committed:
fix the bug where markers name are not show in the field of the form

HristoYonkov committed:
Create useDeleteVendor.ts

HristoYonkov committed:
Create useGetVendors.ts

HristoYonkov committed:
Create usePostVendor.ts

HristoYonkov committed:
Create useUpdateVendor.ts

HristoYonkov committed:
transfer logic for creating a vendor

HristoYonkov committed:
transfer logic for getting vendors

HristoYonkov committed:
Update VendorTableActionsMenu.tsx

1. transfer logic for updating and deleting a vendor
2. call TableActionsMenu component for build a buttons dynamicly

vexleet commented on src/components/features/admin/VendorsTable.tsx:
```diff
@@ -31,20 +47,24 @@ export default function VendorsTable() {
+            items={vendor.markers?.map((marker) => marker.markerName!) || ([] as string[])}
+          />
+        ) : (
+          <Typography>Empty</Typography>
```
```suggestion
          <Typography>-</Typography>
```

vexleet committed:
Update src/components/features/admin/VendorsTable.tsx
